File: solutions/javascript/matrix/1/matrix.js
export class Matrix {
  constructor(matrixString) {
    this.matrix = matrixString
      .split("\n")
      .map(row => row.split(" ").map(Number));
  }

  get rows() {
    return [...this.matrix];
  }

  get columns() {
    if (this.matrix.length === 0) return [];
    return this.matrix[0].map((_, colIndex) => this.matrix.map(row => row[colIndex]));
  }
}

File: solutions/ruby/matrix/1/matrix.rb
=begin
Write your code for the 'Matrix' exercise in this file. Make the tests in
`matrix_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/matrix` directory.
=end

class Matrix
  def initialize(string)
    @matrix = string.split("\n").map! {
      |row| row.to_s.split(" ").map! {
        |number| number.to_i
      }}
  end

  def row(index)
    @matrix.fetch(index - 1)
  end

  def column(index)
    @matrix.transpose.fetch(index - 1)
  end
end
